command/registry.go

```diff
@@ -55,6 +57,19 @@ func setupRegistryForPackerConfig(pc *hcl2template.PackerConfig) hcl.Diagnostics
 		}
 	}
 
+	if env.IsPAREnabled() {
+		hasHCP = true
+	}
+
+	explicit, explicitOK := os.LookupEnv(env.HCPPackerRegistry)
+	if explicitOK && strings.ToLower(explicit) == "off" || explicit == "0" {
+		hasHCP = false
+	}
+
+	if !hasHCP {
+		return nil
+	}
+
```

Comment:
nit: We can take the happy path and remove the need for hasHCP variable. What do you think?

```suggestion
	explicit := os.Getenv(env.HCPPackerRegistry)
	if strings.ToLower(explicit) == "off" || explicit == "0" {
		return nil
	}

	if !env.IsPAREnabled() {
		return nil
	}

```